### PR TITLE
Check proof page UI

### PR DESCRIPTION
**Pull request by ardier16**

### CODE REVIEW

File: src/pages/VerifyProofAlias/index.tsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import { CircularProgress, Stack } from '@mui/material'
+import { useMemo } from 'react'
+import { generatePath, Navigate, useParams } from 'react-router-dom'
+
+import { useLinkProofs } from '@/api'
+import { RoutePaths } from '@/enums'
+
+export default function VerifyProofAlias() {
```
Comment (lukachi) [inline src/pages/VerifyProofAlias/index.tsx:8]: should we create separate place for aliaces, or keep them in pages dir?

File: src/pages/Orgs/pages/OrgsId/pages/OrgCheckProof/components/ProofField.tsx [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+              </UiButton>
+            </InputAdornment>
+          ),
+        }}
+        onChange={e => handleChange(e.target.value)}
+      />
+
+      {!isLoading && isValid !== undefined && (
+        <ProofValidationResult proof={proof} isValid={isValid} />
+      )}
```
[the remark this answers is not in the document]
Comment (ardier16) [reply src/pages/Orgs/pages/OrgsId/pages/OrgCheckProof/components/ProofField.tsx:93]: this way as in the example above `isValid` will be undefined in the prop
```jsx
{isValid !== undefined && <ProofValidationResult proof={proof} isValid={isValid} />}
```

File: src/layouts/MainLayout.tsx [hunk 3]
```diff
@@ -12,6 +12,7 @@ const MainLayout = () => {
         borderRadius={4}
         px={8}
         py={6}
+        overflow={'auto'}
```
Comment (lukachi) [inline src/layouts/MainLayout.tsx:15]: merge main, there should be overflow already

File: src/hooks/loading.ts [hunk 4]
```diff
@@ -61,5 +62,11 @@ export const useLoading = <T>(
     }
   }
 
-  return { data, isLoading, isLoadingError, isEmpty, reload, update }
+  const reset = () => {
+    setIsLoading(false)
+    setIsLoadingError(false)
+    setData(initialState as T)
+  }
```
Comment (lukachi) [inline src/hooks/loading.ts:69]: what if loadOnMount = true? which behaviour expects here?

File: src/pages/Orgs/pages/OrgsId/pages/OrgCheckProof/components/OrgOverview.tsx [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+        {org.status.value === OrgsStatuses.Verified && (
+          <UiIcon componentName={'verified'} size={5} sx={{ color: palette.success.main }} />
+        )}
+      </Stack>
+
+      <Typography mt={2} variant={'body2'}>
+        {org.domain}
+      </Typography>
+
+      <Box mt={6} display={'grid'} gridTemplateColumns={'1fr 1px 1fr'} gap={6} textAlign={'center'}>
```
Comment (lukachi) [inline src/pages/Orgs/pages/OrgsId/pages/OrgCheckProof/components/OrgOverview.tsx:37]: why not Grid component?

File: src/pages/Orgs/pages/OrgsId/pages/OrgCheckProof/components/OrgOverview.tsx [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+        <Box>
+          <Typography variant={'h6'}>{org.members_count}</Typography>
+          <Typography variant={'body2'}>Associated people</Typography>
+        </Box>
+        <Divider orientation={'vertical'} flexItem />
```
Comment (lukachi) [inline src/pages/Orgs/pages/OrgsId/pages/OrgCheckProof/components/OrgOverview.tsx:42]: grid container, but flex item?

File: src/pages/Orgs/pages/OrgsId/pages/OrgCheckProof/components/ProofField.tsx [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+      {
+        [ProofTypes.FirstName]: 'Enter the First Name',
+        [ProofTypes.LastName]: 'Enter the Last Name',
+        [ProofTypes.Age]: 'Enter the Age',
+        [ProofTypes.Telegram]: 'Enter the Telegram Handle',
+      }[proof.proof_type] ?? 'Enter the Details'
+    )
+  }, [proof.proof_type])
+
+  const handleChange = (value: string) => {
```
Comment (lukachi) [inline src/pages/Orgs/pages/OrgsId/pages/OrgCheckProof/components/ProofField.tsx:63]: debounce

File: src/pages/Orgs/pages/OrgsId/pages/OrgCheckProof/components/OrgOverview.tsx (hunk 5 above)
Comment (ardier16) [reply src/pages/Orgs/pages/OrgsId/pages/OrgCheckProof/components/OrgOverview.tsx:37]: it uses flexbox under the hood:
https://mui.com/material-ui/react-grid/#css-grid-layout

File: src/pages/Orgs/pages/OrgsId/pages/OrgCheckProof/components/ProofForm.tsx [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+        InputProps={{
+          startAdornment: (
+            <InputAdornment position='start'>
+              <UiIcon componentName='search' sx={{ color: palette.text.secondary }} />
+            </InputAdornment>
+          ),
+        }}
+      />
+
+      {isLoading || !proofs
```
Comment (lukachi) [inline src/pages/Orgs/pages/OrgsId/pages/OrgCheckProof/components/ProofForm.tsx:32]: isLoadingError

File: src/pages/Orgs/pages/OrgsId/pages/OrgCheckProof/components/ProofValidationResult.tsx [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+              variant={'caption'}
+              // TODO: fix colors
+              color={palette.text.primary}
+              sx={{ overflow: 'hidden', textOverflow: 'ellipsis' }}
+            >
+              {detail.value}
+            </Typography>
+          ) : (
+            detail.value
+          )}
```
[the remark this answers is not in the document]
Comment (lukachi) [reply src/pages/Orgs/pages/OrgsId/pages/OrgCheckProof/components/ProofValidationResult.tsx:76]: let's use value as ReactNode, and use all formattings there

File: src/pages/Orgs/pages/OrgsId/pages/OrgCheckProof/components/ProofField.tsx [hunk 10]
```diff
@@ -0,0 +1,94 @@
...
+
+import { Proof } from '@/api'
+import { sleep } from '@/helpers'
+import { useLoading } from '@/hooks'
+import { UiButton, UiTextField } from '@/ui'
+
+import ProofValidationResult from './ProofValidationResult'
+
+// TODO: fill in a proper way and move to constants
+enum ProofTypes {
```
Comment (lukachi) [inline src/pages/Orgs/pages/OrgsId/pages/OrgCheckProof/components/ProofField.tsx:12]: parse schemas with loadAndParse... method in org svc

File: src/hooks/loading.ts (hunk 4 above)
Comment (ardier16) [reply src/hooks/loading.ts:69]: should be fine

File: src/pages/Orgs/pages/OrgsId/pages/OrgCheckProof/components/ProofField.tsx (hunk 7 above)
Comment (ardier16) [reply src/pages/Orgs/pages/OrgsId/pages/OrgCheckProof/components/ProofField.tsx:47]: IMO not necessary at the moment

File: src/pages/Orgs/pages/OrgsId/pages/OrgCheckProof/components/ProofValidationResult.tsx (hunk 9 above)
[the remark this answers is not in the document]
Comment (ardier16) [reply src/pages/Orgs/pages/OrgsId/pages/OrgCheckProof/components/ProofValidationResult.tsx]: added date formatting

File: src/pages/Orgs/pages/OrgsId/pages/OrgCheckProof/components/ProofField.tsx [hunk 11]
```diff
@@ -0,0 +1,85 @@
+import { InputAdornment, Stack } from '@mui/material'
+import { startCase } from 'lodash'
```
Comment (lukachi) [inline src/pages/Orgs/pages/OrgsId/pages/OrgCheckProof/components/ProofField.tsx:2]: treeshaking issues, let's use:

import startCase from 'lodash/startCase'

File: src/pages/Orgs/pages/OrgsId/pages/OrgCheckProof/components/ProofField.tsx [hunk 12]
```diff
@@ -0,0 +1,85 @@
...
+    setValue(value)
+    reset()
+  }
+
+  // TODO: Handle states properly
+  return isSchemaLoading ? (
+    <></>
+  ) : isSchemaLoadingError ? (
+    <></>
+  ) : isSchemaEmpty || !parsedSchema ? (
```
Comment (lukachi) [inline src/pages/Orgs/pages/OrgsId/pages/OrgCheckProof/components/ProofField.tsx:57]: mb !formattedFieldKey instead of !parsedSchema?

**Review by lukachi — APPROVED** [review]
lgtm

File: src/pages/Orgs/pages/OrgsId/pages/OrgCheckProof/components/ProofField.tsx (hunk 12 above)
Comment (ardier16) [reply src/pages/Orgs/pages/OrgsId/pages/OrgCheckProof/components/ProofField.tsx:57]: yep, forgot to update

**Pull request merged**